### PR TITLE
all methods are done and all dao queries are fixed

### PR DESCRIPTION
**Pull request by Raychenko88**

### CODE REVIEW

File: src/main/java/info/sjd/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE);
+    }
+
+    @GetMapping(path = "find_order_by_item")
+    public ResponseEntity<Order> findOrderByItem(@RequestParam Integer itemId){
+        return new ResponseEntity<>(orderService.findOrderByItem(itemId), HttpStatus.OK);
+    }
+
+    @GetMapping(path = "find_by_cart")
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/OrderController.java]: No reason to use this. It's the same as before.

File: src/main/java/info/sjd/dao/CartDAO.java [hunk 2]
```diff
@@ -3,15 +3,21 @@
...
 import java.util.List;
 
 @Repository
 public interface CartDAO extends JpaRepository<Cart, Integer> {
-    List<Cart> getAllByUserAndPeriod(User user, Long timeFrom, Long timeTo);
 
-    Cart getByUserAndOpenStatus(User user, Integer status);
+    @Query(value = "SELECT * FROM carts WHERE user_id=:userId AND creation_time >=:timeDown AND creation_time <=:timUp",
+            nativeQuery = true)
+    List<Cart> getAllByUserAndPeriod(Integer userId, Long timeDown, Long timUp);
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/dao/CartDAO.java]: Mistake in word timUp

File: src/main/java/info/sjd/dao/CartDAO.java [hunk 3]
```diff
@@ -3,15 +3,21 @@
...
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
 public interface CartDAO extends JpaRepository<Cart, Integer> {
-    List<Cart> getAllByUserAndPeriod(User user, Long timeFrom, Long timeTo);
 
-    Cart getByUserAndOpenStatus(User user, Integer status);
+    @Query(value = "SELECT * FROM carts WHERE user_id=:userId AND creation_time >=:timeDown AND creation_time <=:timUp",
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/dao/CartDAO.java]: Mistake in word timUp

File: src/main/java/info/sjd/dao/CartDAO.java [hunk 4]
```diff
@@ -3,15 +3,21 @@
...
+            nativeQuery = true)
+    List<Cart> getAllByUserAndPeriod(Integer userId, Long timeDown, Long timUp);
 
-    Cart updateStatus(Cart cart, Integer closed);
+    @Query(value = "SELECT * FROM carts WHERE user_id =:id AND closed='0'", nativeQuery = true)
+    Cart getByUserAndOpenStatus(Integer id, Integer status);
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/dao/CartDAO.java]: The second parameter is never used

File: src/main/java/info/sjd/dao/ItemDAO.java [hunk 5]
```diff
@@ -11,7 +12,12 @@
 public interface ItemDAO extends JpaRepository<Item, Integer> {
     List<Item> getByCode(String code);
 
-//    List<Item> getAllByCart(Cart cart);
+    @Query(value = "SELECT ALL FROM items i " +
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/dao/ItemDAO.java]: Wrong query. ALL must be changed to parameters list or *

File: src/main/java/info/sjd/dao/CartDAO.java (hunk 2 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/dao/CartDAO.java]: bug fixed

File: src/main/java/info/sjd/dao/CartDAO.java (hunk 3 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/dao/CartDAO.java]: bug fixed

File: src/main/java/info/sjd/dao/CartDAO.java (hunk 4 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/dao/CartDAO.java]: bug fixed

File: src/main/java/info/sjd/dao/ItemDAO.java (hunk 5 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/dao/ItemDAO.java]: bug fixed

File: src/main/java/info/sjd/controller/OrderController.java (hunk 1 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/OrderController.java]: bug fixed

File: src/main/java/info/sjd/controller/CartController.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    public ResponseEntity<List> getAllByUserAndPeriod(@RequestParam Integer userId, Long timeDown, Long timeUp){
+        return new ResponseEntity<>(cartService.getAllByUserAndPeriod(userId, timeDown, timeUp), HttpStatus.OK);
+    }
+
+    @GetMapping(path = "open_status")
+    public ResponseEntity<Cart> getByUserAndOpenStatus(@RequestParam Integer id){
+        return new ResponseEntity<>(cartService.getByUserAndOpenStatus(id), HttpStatus.OK);
+    }
+
+    @GetMapping(path = "update_status")
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/CartController.java]: Methid must be post

File: src/main/java/info/sjd/controller/CartController.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    public ResponseEntity<Cart> updateStatus(@RequestParam Integer idParam, Integer closedParam){
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/CartController.java]: Add annotation to the second parameter

File: src/main/java/info/sjd/controller/CartController.java (hunk 6 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/CartController.java]: bug fixed

File: src/main/java/info/sjd/controller/CartController.java (hunk 7 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/CartController.java]: bug fixed

**Review by Ignatenko2207 — APPROVED** [review]

**Pull request merged**